api/src/main/java/org/openmrs/api/impl/OrderServiceImpl.java

```diff
@@ -337,9 +337,13 @@ private void discontinueExistingOrdersIfNecessary(Order order, Boolean isRetrosp
 			return;
 		}
 
+		Date asOfDate = null;
+		if (isRetrospective)
+			asOfDate = order.getDateActivated();
+
 		//Mark first order found corresponding to this DC order as discontinued.
 		List<? extends Order> orders = getActiveOrders(order.getPatient(), order.getOrderType(), order.getCareSetting(),
-		    null);
+		    asOfDate);
```

Comment:
@wluyima If a user wants to record the fact that a discontinue order was issued on a certain day, how would he/she do it?